lib/api/core/models/repositories/profileRepository.js

```diff
@@ -7,251 +9,252 @@ var
   NotFoundError = require('kuzzle-common-objects').errors.NotFoundError,
   Repository = require('./repository');
 
-
-function extractRoleIds(policiesOrRoles) {
-  return policiesOrRoles.map(element => {
-    if (element.roleId) {
-      return element.roleId;
-    }
-    return element._id;
-  });
-}
-
 /**
  * @class ProfileRepository
  * @extends Repository
- * @constructor
- */
-function ProfileRepository () {
-  Repository.apply(this, arguments);
-  this.collection = 'profiles';
-  this.ObjectConstructor = Profile;
-  this.profiles = {};
-}
-
-ProfileRepository.prototype = Object.create(Repository.prototype);
-ProfileRepository.prototype.constructor = ProfileRepository;
-
-ProfileRepository.prototype.init = function profileRepositoryInit () {
-  Repository.prototype.init.call(this, {});
-};
-
-/**
- * Loads a Profile object given its id.
- *
- * @param {string} profileId
- * @returns {Promise} Resolves to the matching Profile object if found, null if not.
  */
-ProfileRepository.prototype.loadProfile = function profileRepositoryLoadProfile (profileId) {
-  if (!profileId) {
-    return Promise.reject(new BadRequestError('Missing profileId'));
+class ProfileRepository extends Repository {
+  /**
+   * @params {Kuzzle} kuzzle
+   * @constructor
+   */
+  constructor (kuzzle) {
+    super(kuzzle);
+
+    this.collection = 'profiles';
+    this.ObjectConstructor = Profile;
+    this.profiles = {};
   }
 
-  if (typeof profileId !== 'string') {
-    return Promise.reject(new BadRequestError(`Invalid argument: Expected profile id to be a string, received "${typeof profileId}"`));
+  init () {
```

Comment:
the override is not used to define a default to options, but to force it to an empty object.